src/installer/destinations.py

```diff
@@ -176,20 +208,32 @@ def write_script(
         with io.BytesIO(data) as stream:
             entry = self.write_to_fs(Scheme("scripts"), script_name, stream)
 
-            path = os.path.join(self.scheme_dict[Scheme("scripts")], script_name)
+            path = self._destdir_path(Scheme("scripts"), script_name)
             mode = os.stat(path).st_mode
             mode |= (mode & 0o444) >> 2
             os.chmod(path, mode)
 
             return entry
 
+    def _compile_bytecode(self, scheme: Scheme, record: RecordEntry) -> None:
+        """Compile bytecode for a single .py file."""
+        if scheme not in ("purelib", "platlib"):
```

Comment:
For a follow up: we should consider what the right behaviour for `scripts` would be.

Reply:
AFAIK, Python doesn't even look for cached bytecode for a script it's running, only on imports. Of course, you could have a script that is also importable, but I wouldn't spend too much time thinking about that - the lib directories are the obvious place to put code to be imported.

Reply:
Ah, no worries then. :)